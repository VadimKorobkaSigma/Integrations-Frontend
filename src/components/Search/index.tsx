import React from 'react';
import cn from 'classnames';

import { normalizeString } from '@services/utils';
import Input, { InputProps } from '@components/Input';
import searchIcon from '@assets/images/search.svg';
import styles from './styles.module.scss';

interface HighlightProps {
    searchQuery: string;
    text: string;
}

export const SearchHighlight: React.FC<HighlightProps> = React.memo(({ searchQuery, text }) => {
    if (!searchQuery) {
        return <span>{text}</span>;
    }

    const highlight = normalizeString(searchQuery);
    const parts = highlight.length ? text.split(new RegExp(`(${highlight})`, 'gi')) : [text];
    return (
        <span>
            {parts.map((part, i) => (
                <span
                    key={i}
                    className={cn({
                        [styles.highlighted]: normalizeString(part.toLowerCase()) === highlight.toLowerCase(),
                    })}
                >
                    {part}
                </span>
            ))}
        </span>
    );
});

SearchHighlight.displayName = 'Search.Highlight';

const Search: React.FC<InputProps> = (props) => {
    return <Input icon={searchIcon} placeholder="Search" {...{ ...props }} />;
};

export default React.memo(Search);
