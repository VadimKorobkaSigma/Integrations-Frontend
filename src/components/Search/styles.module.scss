.highlighted {
  background-color: #{$green}60;
}
