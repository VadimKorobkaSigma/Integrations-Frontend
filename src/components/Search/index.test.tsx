import * as React from 'react';
import Search, { SearchHighlight } from './index';
import { shallow } from 'enzyme';

test('Search Component without error', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper.html()).toMatchSnapshot();
});

test('SearchHighlight Component without error', () => {
    const wrapper = shallow(<SearchHighlight text="Test name" searchQuery="Test" />);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ searchQuery: '' });
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ text: '', searchQuery: 'Test' });
    expect(wrapper.html()).toMatchSnapshot();
});
