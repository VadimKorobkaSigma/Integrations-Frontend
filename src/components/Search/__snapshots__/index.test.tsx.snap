// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component without error 1`] = `"<div class=\\"\\"><input placeholder=\\"Search\\" class=\\"\\"/></div>"`;

exports[`SearchHighlight Component without error 1`] = `"<span><span class=\\"\\"></span><span class=\\"undefined\\">Test</span><span class=\\"\\"> name</span></span>"`;

exports[`SearchHighlight Component without error 2`] = `"<span>Test name</span>"`;

exports[`SearchHighlight Component without error 3`] = `"<span><span class=\\"\\"></span></span>"`;
